### PR TITLE
Feature/task30 - Implement Spring Security

### PR DESCRIPTION
**Pull request by TiagoVidal91**

### CODE REVIEW

File: src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Security/Service/RoleServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                .collect(Collectors.toSet());
+    }
+    @Override
+    public Optional<Role> findByName(String name) {
+        return roleRepo.findByName(name);
+    }
+    private Role safeFindByName(String roleStr) {
+        return findByName(roleStr)
+                .orElseThrow(() -> new RuntimeException("Error: Role is not found."));
+    }
```
Comment (clara-a-correia) [inline src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Security/Service/RoleServiceImpl.java:30]: Consider condensing methods

File: src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Configurations/WebSecurityConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Test
+    @WithMockUser(authorities = {"ADMIN"})
+    public void testAdminAccess() throws Exception {
+        mockMvc.perform(get("/teiai-api/admin/test/test-admin-response"))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void testUserAccess() throws Exception {
```
Comment (clara-a-correia) [inline src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Configurations/WebSecurityConfigTest.java:48]: Test user access on admin page

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -8,3 +8,9 @@ spring.jpa.properties.hibernate.format_sql=true
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
 
 spring.security.enabled=false
+
```
Comment (clara-a-correia) [inline src/main/resources/application.properties]: Change secret, add file to gitignore and remove it from PR

**Review by clara-a-correia — CHANGES_REQUESTED** [review]

File: src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Security/Service/RoleServiceImpl.java (hunk 1 above)
Comment (TiagoVidal91) [reply src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Security/Service/RoleServiceImpl.java:30]: Done

File: src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Configurations/WebSecurityConfigTest.java (hunk 2 above)
Comment (TiagoVidal91) [reply src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Configurations/WebSecurityConfigTest.java:48]: Done

File: src/main/resources/application.properties (hunk 3 above)
Comment (TiagoVidal91) [reply src/main/resources/application.properties]: Done

**Review by clara-a-correia — APPROVED** [review]

**Pull request merged**